python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -265,6 +283,40 @@ def _legacy_worker_node_to_launch(
             else:
                 return {}
 
+    def _update_node_resources_from_runtime(
+            self, nodes: List[NodeID],
+            max_resources_by_ip: Dict[NodeIP, ResourceDict]):
+        """Update static node type resources with runtime resources
+
+        This will update the cached static node type resources with the runtime
+        resources. Because we can not know the correctly memory or
+        object_store_memory from config file.
+        """
+        need_update = len(self.node_types) != len(self.node_resource_updated)
+
+        if not need_update:
+            return
+        for node_id in nodes:
+            tags = self.provider.node_tags(node_id)
+
+            if TAG_RAY_USER_NODE_TYPE not in tags:
+                continue
+
+            node_type = tags[TAG_RAY_USER_NODE_TYPE]
+            if (node_type in self.node_resource_updated
+                    or node_type not in self.node_types):
+                # continue if the node type has been updated or is not an known
+                # node type
+                continue
+            ip = self.provider.internal_ip(node_id)
+            runtime_resources = max_resources_by_ip.get(ip)
+            if runtime_resources:
+                runtime_resources = copy.deepcopy(runtime_resources)
+                resources = self.node_types[node_type].get("resources", {})
+                resources.update(runtime_resources)
```

Comment:
Can we hardcode this to only copy cpu, gpu, memory, and object_store_memory? I think this will also try to copy over custom resources (like placement groups) which will have unintended consequences.

Reply:
OK, will update it.